runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/PortablePipelineResult.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.fnexecution.jobsubmission;
+
+import org.apache.beam.model.jobmanagement.v1.JobApi;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.annotations.Experimental;
+
+/** Result of a portable {@link Pipeline#run()}. */
+public interface PortablePipelineResult extends PipelineResult {
+
+  /**
+   * Returns the object to access monitoring infos from the pipeline.
+   *
+   * @throws UnsupportedOperationException if the runner doesn't support retrieving metrics.
+   */
+  @Experimental(Experimental.Kind.METRICS)
+  JobApi.MetricResults monitoringInfos() throws UnsupportedOperationException;
```

Comment:
portableMetrics() sounds fine. It was the name in the first version but I changed because I was unsure which version is more descriptive. Now that you brought that I think there's no sense to introduce new term (monitoringInfos) to the interface but only distinguish with the "portable" prefix (tldr: I agree :) ).